src/Core/Layer/Layer.js

```diff
@@ -1,3 +1,22 @@
+import { EventDispatcher } from 'three';
+
+
+export const defineLayerProperty = function defineLayerProperty(layer, propertyName, defaultValue) {
+    var property = layer[propertyName] == undefined ? defaultValue : layer[propertyName];
+    Object.defineProperty(layer,
+        propertyName,
+        { get: () => property,
+            set: (newValue) => {
+                if (property !== newValue) {
+                    const event = { type: 'layer-changed', previous: {}, new: {} };
+                    event.previous[propertyName] = property;
+                    event.new[propertyName] = newValue;
+                    property = newValue;
+                    layer.dispatchEvent(event);
+                }
+            } });
+};
```

Comment:
Atlernative proposal:
```js
export const defineLayerProperty = function defineLayerProperty(layer, propertyName, defaultValue, onChange) {
    var property = layer[propertyName] == undefined ? defaultValue : layer[propertyName];
    Object.defineProperty(layer,
        propertyName,
        {
            get: () => property,
            set: (newValue) => {
                if (property !== newValue) {
                    const event = { type: 'layer-changed', previous: {}, new: {} };
                    event.previous[propertyName] = property;
                    event.new[propertyName] = newValue;
                    property = newValue;
                    if (onChange) {
                        onChange(layer, propertyName);
                    }
                    layer.dispatchEvent(event);
                }
            },
        });
};
```
I don't like public events (`layer-changed` here) being used for internal changes (calling `_syncThreejsLayer`) because if a user listen for the same event, you can't garantee the calling order.
So my proposal introduces a simple `onChange` callback to cover this. 
WDYT?